hangman.py

```diff
@@ -0,0 +1,222 @@
+
+wordlist = load_words()
+
+
+def is_word_guessed(secret_word, letters_guessed):
+    '''
+    secret_word: string, the word the user is guessing; assumes all letters are
+      lowercase
+    letters_guessed: list (of letters), which letters have been guessed so far;
+      assumes that all letters are lowercase
+    returns: boolean, True if all the letters of secret_word are in letters_guessed;
+      False otherwise
+    '''
+    for i in range(len(secret_word)):
+        if secret_word[i] not in letters_guessed:
+            return False
+    return True
+
+
+def get_guessed_word(secret_word, letters_guessed):
+    '''
+    secret_word: string, the word the user is guessing
+    letters_guessed: list (of letters), which letters have been guessed so far
+    returns: string, comprised of letters, underscores (_), and spaces that represents
+      which letters in secret_word have been guessed so far.
+    '''
+    splited_word = list(secret_word)
+    for i in range(len(secret_word)):
+        if secret_word[i] not in letters_guessed:
+            splited_word[i] = LOWOER_SPACE + " "
+    return "".join(splited_word)
+
+
+def get_available_letters(letters_guessed):
+    '''
+    letters_guessed: list (of letters), which letters have been guessed so far
+    returns: string (of letters), comprised of letters that represents which letters have not
+      yet been guessed.
+    '''
+    avaible_letters = string.ascii_lowercase
+    alphabet = filter(lambda letter: letter not in letters_guessed, string.ascii_lowercase)
+    return ''.join(alphabet)
+
+
+def hangman(secret_word, hints=False):
+    '''
+    secret_word: string, the secret word to guess.
+    
+    Starts up an interactive game of Hangman.
+    
+    * At the start of the game, let the user know how many 
+      letters the secret_word contains and how many guesses s/he starts with.
+      
+    * The user should start with 6 guesses
+
+    * Before each round, you should display to the user how many guesses
+      s/he has left and the letters that the user has not yet guessed.
+    
+    * Ask the user to supply one guess per round. Remember to make
+      sure that the user puts in a letter!
+    
+    * The user should receive feedback immediately after each guess 
+      about whether their guess appears in the computer's word.
+
+    * After each guess, you should display to the user the 
+      partially guessed word so far.
+    
+    Follows the other limitations detailed in the problem write-up.
+    '''
+    warnings = 3
+    attemps = 6
+    letters_guessed = set()
+    print(f'I am thinking of a word that is {len(secret_word)} letters long')
+    print(f'Your have {warnings} warnings')
+    while attemps > 0 and not is_word_guessed(secret_word, letters_guessed):
+        print("-" * 13)
+        print("You have ", attemps, " guesses left.")
+        print("Available letters: ", get_available_letters(letters_guessed))
+        letter = input("enter letter:").strip(' ')
```

Comment:
Have you been checked if output of your game is the same as output in task examples?